src/api/v1/permission/permission.model.ts

```diff
@@ -24,8 +24,24 @@ const PermissionSchema: Schema<Permission> = new Schema<Permission>(
       enum: Object.values(PermissionCategory),
       required: true,
     },
-    createdBy: Types.ObjectId,
-    updatedBy: Types.ObjectId,
+    createdBy: {
+      type: String,
+      required: true,
+      validate: {
+        validator: (v: string) => Types.ObjectId.isValid(v),
+        message: 'createdBy must be a valid MongoDB ObjectId string',
+      },
+      ref: 'User',
+    },
+    updatedBy: {
+      type: String,
+      required: true,
+      validate: {
+        validator: (v: string) => Types.ObjectId.isValid(v),
+        message: 'updatedBy must be a valid MongoDB ObjectId string',
+      },
+      ref: 'User',
+    },
```

Comment:
@sanjaysah101, we can simplify the system by removing the `createdBy` and `updatedBy` fields for roles and permissions. Since we'll be using static roles and permissions defined as constants, there's no need to create new roles or permissions from the admin panel. This approach reduces complexity and ensures a more robust service. The admin panel will only allow modifications to the relationships between roles and permissions, and between users and roles. By keeping predefined roles and permissions in the database, we can maintain a clear and predictable flow in the application. On the client side, we can dynamically adjust behaviour when permissions are assigned or removed from roles.

Reply:
Hi @ansopedia, that sounds great! However, I believe we still need to implement the CRUD operations for roles. Specifically, we should allow the admin to create new roles from the admin panel and assign predefined permissions to those roles.

For example, if we have a team and the admin wants to create a new role and assign it to a user, this will help us build a more flexible RBAC system. Additionally, our API flow remains predictable since routes are protected by specific permissions, and users have roles that are tied to those permissions. If we need to grant or revoke access in the future, we can simply modify the associated role, making the system adaptable and easy to manage.